### PR TITLE
Cause Model Template: outline and getting started instructions

### PR DESCRIPTION
**Pull request by NathanielBlairStahn**
Please review the section titles and the order of the sections, and post here if you want to suggest changes. I've put a few specific questions/concerns in special "Todo for template development" boxes within the .rst document itself.

### CODE REVIEW

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    Are the sections in a good order?
+
+Model Assumptions and Limitations
+---------------------------------
+
+Restrictions
+++++++++++++
+
+Scope
++++++
```
Comment (XiaoluQianUW) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: The template looks good to me. I was putting the model assumptions and limitations section right after the cause model diagram. But I think this order makes more sense.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+Disease Description
+-------------------
+
+.. todo::
+
+   Add description of cause.
+
+.. admonition:: Todo for template development
+
+   - Confirm the name of this section. Will "Disease Description" be inaccurate
```
Comment (alibow) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: I think either is fine as long as we clarify somewhere that this should be a general clinical overview of the cause.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 1 above)
Comment (yongqx) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Model assumptions and/or limitations can be mentioned  in summary disease model description and fully explained in `Model Assumptions and Limitations` section.

**Comment by yongqx** [comment]
We'd better point it out that editable svg format of fig is preferable and one data-relevant section needed for more complicated model is non-standard data sources where external calculation occurs.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 2 above)
Comment (collijk) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: You could drop the subsection title all together and put the description at the top level.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+     or confusing if we are modeling a cause that is not technically a disease?
+     I was aiming for something that matched everyday usage better than
+     "Cause Description."
+
+   - In the above todo box:
+
+     - Add more details about what we're looking for in cause descriptions
+
+   - For the next section, is it important for the cause name to appear in the
+     section title, as it currently does, or should we call the section
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: I don't think it needs the cause name.  Something like `GBD Modeling Strategy` seems fine.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+   #. Once you complete these steps, delete this :code:`.. important::`
+      `directive <https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#directives>`_
+      from :file:`index.rst`.
+
+Disease Description
+-------------------
+
+.. todo::
+
+   Add description of cause.
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: Instructions here should include:
- Useful external data sources
- note to flesh out how this cause kills or causes disability among the with condition
- Important features of the cause (vaccine coverage, is it a progressive disease, etc.)
- Links to other prominent mathematical models of the cause if they exist (e.g. @yongqx found like 40 different versions of tb models).

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+   **Note:** This `custom admonition
+   <https://docutils.sourceforge.io/docs/ref/rst/directives.html#generic-admonition>`_
+   indicates tasks to do while we write this template, whereas the ordinary
+   Todo's (and the "Important" directives) are intended to remain in the
+   template to instruct the cause modeler how to fill out the cause model
+   document.
+
+Modeling `{Cause Name}` in GBD 2017
+-----------------------------------
+
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: Point people to yld and cod appendices for their diseases and link to http://ghdx.healthdata.org/gbd-2017/code which can serve as an annoying to read but authoritative source on how the cause was actually modeled.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      <https://docutils.sourceforge.io/docs/user/rst/quickref.html#section-structure>`_
+      when Sphinx builds HTML from the :file:`index.rst` file.
+
+   #. Once you complete these steps, delete this :code:`.. important::`
+      `directive <https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#directives>`_
+      from :file:`index.rst`.
+
+Cause Hierarchy
++++++++++++++++
+
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: To fill in: description of our strategy for making cause hierarchy diagrams (rules + example)

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+Cause Model Diagram
+-------------------
+
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: To fill in: description of what the bubbles and arrows represent.  Maybe include svg templates for common diagrams like SI, SIS, SIR, etc.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+State and Transition Data Tables
+--------------------------------
+
+.. admonition:: Todo for template development
+
+    Decide on section heading names and structure. E.g. should this section be
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: I think `Data Description` is fine.

File: docs/source/models/gbd2017/causes/cause_model_template.rst [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    called "Data Description" instead of "State and Transition Data Tables,"
+    like it was originally? Is there anything else that belongs in a "Data
+    Description" section besides the data tables?
+
+    Are the sections in a good order?
+
+Model Assumptions and Limitations
```
Comment (collijk) [inline docs/source/models/gbd2017/causes/cause_model_template.rst]: I think the restrictions in this section should move up to the GBD Modeling section.

Also, perhaps we should have a section following the GBD section called `vivarium modeling strategy` which would include the scope and the restrictions we apply to the model (which might be different than GBD's restrictions).  To serve as a narrative description accompanying the cause model diagram and data tables.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 6 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Coming in a later pull request - this is just the outline

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 7 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Coming in a later pull request - this is just the outline

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 5 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Thanks for the link, @collijk  - I'll add that. I'll be adding references to the appendices in a later pull request.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 4 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Thanks for the suggestions, @collijk , I'll add those.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 3 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Seems reasonable

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 2 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: > You could drop the subsection title all together and put the description at the top level.

Yeah, perhaps that's better.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 9 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: Ok, that seems reasonable. What do other people think?

Do we have any examples of restrictions we would apply that are different from GBD restrictions?

**Comment by NathanielBlairStahn** [comment]
@collijk suggested several changes, whereas several people (including @collijk ) have approved the current version.

Should I merge the current version in, and make another PR to implement those changes so people have a chance to discuss?

Or should I push the changes to this branch before merging?

@ihmeuw/vivarium-research

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 1 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: @yongqx , would James's suggestion below align with what you're thinking here?

> I think the restrictions in this section should move up to the GBD Modeling section.
> 
> Also, perhaps we should have a section following the GBD section called `vivarium modeling strategy` which would include the scope and the restrictions we apply to the model (which might be different than GBD's restrictions). To serve as a narrative description accompanying the cause model diagram and data tables.

**Comment by NathanielBlairStahn** [comment]
> I am good with this structure. Bu, we have to make changes to the causes that are finished. Also, for restrictions I like it under Assumptions and Limitations section. We can add subsections if there are different types of restrictions.

Thanks @kiranklc . Yes, we will have to modify the cause documents that are already finished once we agree on a finalized version of this template.  I think co-writing this template is where we will work out what the best structure is, based on everyone's experience so far.

I'm going to leave the structure as is for now but make note of everyone's comments in the todo's before merging into master. I think I will open another pull request implementing some of @collijk 's ideas, so if you don't like how that looks, you can add further comments in the new pull request once I open it.

File: docs/source/models/gbd2017/causes/cause_model_template.rst (hunk 2 above)
Comment (NathanielBlairStahn) [reply docs/source/models/gbd2017/causes/cause_model_template.rst]: > I think either is fine as long as we clarify somewhere that this should be a general clinical overview of the cause.

Duly noted. I will edit the todo.

**Pull request merged**